lib/package.py

```diff
@@ -180,3 +184,31 @@ def _download_build_files(self):
             filename = os.path.join(self.build_files, url.split('/')[-1])
             with open(filename, "wb") as file_data:
                 file_data.write(data)
+
+    def lock(self):
+        """
+        Locks the package to avoid concurrent operations on its shared
+        resources.
+        Currently, the only resource shared among scripts executed from
+        different directories is the repository.
+        """
+        LOG.debug("Checking for lock on file {}.".format(self.lock_file_path))
+        self.lock_file = open(self.lock_file_path, "w")
+        try:
+            fcntl.lockf(self.lock_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
IMHO locking the source code repository when someone is using it is not an optimal solution. This only allows the simultaneous build of one package at once and only parallelizes builds of different packages. I'd prefer to use a  solution which parallelizes builds of *any* package , like using a dictionary (repo_name, commit_id) -> repo_directory_path